### PR TITLE
Properly update the secrets of the authorizer

### PR DESCRIPTION
**Pull request by dstiliadis**
Update the secrets of the API authorizer when they expire.

### CODE REVIEW

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1497",
    "commit-sha": "3a556f1230c1a1936e9a07eec46dcc2ed6d5b129"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "916",
    "commit-sha": "024d5e87ca15f0e9857c6c3b32b1eff48e883a95"
  }
]
```

File: controller/internal/enforcer/applicationproxy/http/http.go [hunk 1]
```diff
@@ -329,15 +329,16 @@ func (p *Config) ShutDown() error {
...
 	p.Lock()
-	defer p.Unlock()
-
 	p.cert = cert
 	p.ca = caPool
 	p.secrets = s
 	p.certPEM = certPEM
 	p.keyPEM = keyPEM
 	p.tlsClientConfig.RootCAs = caPool
+	p.Unlock()
```
Comment (abhijitherekar) [inline controller/internal/enforcer/applicationproxy/http/http.go:338]: the earlier defer p.Unlock also would work as the apiAuth has its own lock.

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "1497",
    "commit-sha": "3a556f1230c1a1936e9a07eec46dcc2ed6d5b129"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "916",
    "commit-sha": "71bb491c0e8e2523de63bb40e6b75d4779e20359"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "api-auth-secrets",
    "component": "trireme-lib",
    "pr-id": "916",
    "commit-sha": "71bb491c0e8e2523de63bb40e6b75d4779e20359"
  },
  {
    "project": "api-auth-secrets",
    "component": "enforcerd",
    "pr-id": "1502",
    "commit-sha": "02e7dcb86543de2fcc77f581ebd82d18e61db6a6"
  }
]
```

**Review by abhijitherekar — APPROVED** [review]

**Pull request merged**